src/main/java/com/blackducksoftware/integration/hub/alert/channel/DistributionChannel.java

```diff
@@ -71,36 +71,45 @@ public CommonDistributionRepository getCommonDistributionRepository() {
 
     public void setAuditEntrySuccess(final Long auditEntryId) {
         if (auditEntryId != null) {
-            final AuditEntryEntity auditEntryEntity = getAuditEntryRepository().findOne(auditEntryId);
-            if (auditEntryEntity != null) {
-                auditEntryEntity.setStatus(StatusEnum.SUCCESS);
-
-                auditEntryEntity.setTimeLastSent(new Date(System.currentTimeMillis()));
-                getAuditEntryRepository().save(auditEntryEntity);
+            try {
+                final AuditEntryEntity auditEntryEntity = getAuditEntryRepository().findOne(auditEntryId);
+                if (auditEntryEntity != null) {
+                    auditEntryEntity.setStatus(StatusEnum.SUCCESS);
+                    auditEntryEntity.setErrorMessage(null);
+                    auditEntryEntity.setErrorStackTrace(null);
+                    auditEntryEntity.setTimeLastSent(new Date(System.currentTimeMillis()));
+                    getAuditEntryRepository().save(auditEntryEntity);
+                }
+            } catch (final Exception e) {
+                logger.error(e.getMessage(), e);
             }
         }
     }
 
-    public void setAuditEntryFailure(final Long auditEntryId, final String errorMessage, final Throwable e) {
+    public void setAuditEntryFailure(final Long auditEntryId, final String errorMessage, final Throwable t) {
         if (auditEntryId != null) {
-            final AuditEntryEntity auditEntryEntity = getAuditEntryRepository().findOne(auditEntryId);
-            if (auditEntryEntity != null) {
-                auditEntryEntity.setStatus(StatusEnum.FAILURE);
-                auditEntryEntity.setErrorMessage(errorMessage);
-                final String[] rootCause = ExceptionUtils.getRootCauseStackTrace(e);
-                String exceptionStackTrace = "";
-                for (final String line : rootCause) {
-                    if (exceptionStackTrace.length() + line.length() < 9999) {
-                        exceptionStackTrace = exceptionStackTrace + line + System.lineSeparator();
-                    } else {
-                        break;
+            try {
+                final AuditEntryEntity auditEntryEntity = getAuditEntryRepository().findOne(auditEntryId);
+                if (auditEntryEntity != null) {
+                    auditEntryEntity.setStatus(StatusEnum.FAILURE);
+                    auditEntryEntity.setErrorMessage(errorMessage);
+                    final String[] rootCause = ExceptionUtils.getRootCauseStackTrace(t);
+                    String exceptionStackTrace = "";
+                    for (final String line : rootCause) {
+                        if (exceptionStackTrace.length() + line.length() < 9999) {
```

Comment:
Is there a database constant for VARCHAR that we could use here?

Reply:
We set the limit in the AuditEntryEntity. We could use a static variable for the limit so we dont have to change it in both places